lib/net.js

```diff
@@ -307,6 +313,7 @@ function Socket(options) {
   this[BYTES_READ] = 0;
 }
 util.inherits(Socket, stream.Duplex);
+StreamBase.apply(Socket.prototype);
```

Comment:
Can we get rid of `apply` and use `Object.assign()`? I find it cleaner.

```js
Object.assign(Socket.prototype, StreamBase);
```

I also find the `StreamBase` name a bit odd for a set of mixins.

Reply:
yea, i agree. i called it `StreamBaseMethods` prior to this... maybe `StreamSharedMethods`? and +1 on using `Object.assign` will do soon